src/DynamoUtilities/PathHelper.cs

```diff
@@ -83,5 +84,32 @@ public static bool HasWritePermissionOnDir(string folderPath)
 
             return writeAllow && !writeDeny;
         }
+
+        /// <summary>
+        /// searchs for a file with an extension in a list of paths, returns the first match
+        /// where the extension is case insensitive. If no file is found, returns an empty string.
+        /// </summary>
+        /// <param name="filename"></param>
+        /// <param name="extension"></param>
+        /// <param name="searchPaths"></param>
+        /// <returns></returns>
+        public static string FindFileInPaths(string filename, string extension, string[] searchPaths)
+        {
+            foreach (var path in searchPaths)
+            {
+                if (Directory.Exists(path))
+                {
+                    var files = Directory.GetFiles(path);
+                    var matches = files.ToList().Where(x => x.EndsWith(filename+extension) || x.EndsWith(filename+extension.ToUpper()) || x.EndsWith(filename+extension.ToLower()));
+                    if (matches.Count() > 0)
```

Comment:
Matching that is the only reason to replace what you had before with this.
Something.xml, Something.XML, Something.Xml, Something.xmL, they're all the same file on Windows.